turtle_tf2_cpp/src/turtle_tf2_message_filter.cpp

```diff
@@ -0,0 +1,96 @@
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <geometry_msgs/msg/point_stamped.hpp>
+#include <message_filters/subscriber.h>
+
+#include <rclcpp/rclcpp.hpp>
+#include <tf2_ros/buffer.h>
+#include <tf2_ros/create_timer_ros.h>
+#include <tf2_ros/message_filter.h>
+#include <tf2_ros/transform_listener.h>
+#ifdef TF2_CPP_HEADERS
+  #include <tf2_geometry_msgs/tf2_geometry_msgs.hpp>
+#else
+  #include <tf2_geometry_msgs/tf2_geometry_msgs.h>
+#endif
+
+#include <chrono>
+#include <memory>
+#include <string>
+
+using std::placeholders::_1;
+using namespace std::chrono_literals;
+
+class PoseDrawer : public rclcpp::Node
+{
+public:
+  PoseDrawer()
+  : Node("turtle_tf2_pose_drawer")
+  {
+    // Declare and acquire `target_frame` parameter
+    this->declare_parameter<std::string>("target_frame", "turtle1");
+    this->get_parameter("target_frame", target_frame_);
+
+    typedef std::chrono::duration<int> seconds_type;
+    seconds_type buffer_timeout(10);
+
+    tf2_buffer_ = std::make_shared<tf2_ros::Buffer>(this->get_clock());
+    // set timer interface before call to waitForTransform,
+    // to avoid a tf2_ros::CreateTimerInterfaceException is thrown.
+    auto timer_interface = std::make_shared<tf2_ros::CreateTimerROS>(
+      this->get_node_base_interface(),
+      this->get_node_timers_interface());
+    tf2_buffer_->setCreateTimerInterface(timer_interface);
+    tf2_listener_ =
+      std::make_shared<tf2_ros::TransformListener>(*tf2_buffer_);
+    point_sub_.subscribe(this, "/turtle3/turtle_point_stamped");
+    tf2_filter_ = std::make_shared<tf2_ros::MessageFilter<geometry_msgs::msg::PointStamped>>(
+      point_sub_, *tf2_buffer_, target_frame_, 100, this->get_node_logging_interface(),
+      this->get_node_clock_interface(), buffer_timeout);
+    tf2_filter_->registerCallback(&PoseDrawer::msgCallback, this);
+  }
+
+// Callback to register with tf2_ros::MessageFilter to be called when transforms are available
+
+private:
+  void msgCallback(const geometry_msgs::msg::PointStamped::SharedPtr point_ptr)
+  {
+    geometry_msgs::msg::PointStamped point_out;
+    try {
+      tf2_buffer_->transform(*point_ptr, point_out, target_frame_);
+      RCLCPP_INFO(
+        this->get_logger(), "point of turtle 3 in frame of turtle 1 Position(x:%f y:%f z:%f)\n",
+        point_out.point.x,
+        point_out.point.y,
+        point_out.point.z);
+    } catch (tf2::TransformException & ex) {
+      RCLCPP_WARN(
+        this->get_logger(), "Failure %s\n", ex.what());    // Print exception which was caught
```

Comment:
```suggestion
        this->get_logger(), "Failure %s\n", ex.what()); // Print exception which was caught
```

Reply:
@kurshakuz  Here has 4 whitespace,  I removed 2 whitespaces, and left 2 whitespaces, I think it need 2 whitespaces between comments and code, otherwise it will get wrong when linting.

Reply:
@kurshakuz  To ensure it can pass the test and linting, I moved the comment line to a new line before the code line.